src/NServiceBus.Transport.RabbitMQ/RabbitMQTransport.cs

```diff
@@ -9,17 +16,256 @@
+        /// </summary>
+        public TimeSpan TimeToWaitBeforeTriggeringCircuitBreaker
+        {
+            get => timeToWaitBeforeTriggeringCircuitBreaker;
+            set
+            {
+                Guard.AgainstNegativeAndZero("value", value);
+                timeToWaitBeforeTriggeringCircuitBreaker = value;
+            }
+        }
+
+        /// <summary>
+        /// The calculation method for prefetch count. By default 3 times the maximum concurrency value.
+        /// 
+        /// The argument for the callback is the maximum concurrency. The result needs to be a positive integer value.
+        /// </summary>
+        public Func<int, int> PrefetchCountCalculation
+        {
+            get => prefetchCountCalculation;
+            set
+            {
+                Guard.AgainstNull("value", value);
+                prefetchCountCalculation = value;
+            }
+        }
+
+        /// <summary>
+        /// Configures if the client should use TLS-secured connection.
+        /// </summary>
+        public bool UseTLS { get; set; }
+
+        /// <summary>
+        /// The certificate to use for client authentication when connecting to the broker via TLS.
+        /// </summary>
+        public X509Certificate2 ClientCertificate { get; set; }
+
+        /// <summary>
+        /// Should the client validate the broker certificate when connecting via TLS.
+        /// </summary>
+        public bool ValidateRemoteCertificate { get; set; } = true;
+
+        /// <summary>
+        /// Specifies if an external authentication mechanism should be used for client authentication.
+        /// </summary>
+        public bool UseExternalAuthMechanism { get; set; } = false;
+
+        /// <summary>
+        /// The interval for heartbeats between the endpoint and the broker.
+        /// </summary>
+        public TimeSpan HeartbeatInterval
+        {
+            get => heartbeatInterval;
+            set
+            {
+                Guard.AgainstNegativeAndZero("value", value);
+                heartbeatInterval = value;
+            }
+        }
+
+        /// <summary>
+        /// The time to wait between attempts to reconnect to the broker if the connection is lost.
+        /// </summary>
+        public TimeSpan NetworkRecoveryInterval
+        {
+            get => networkRecoveryInterval;
+            set
+            {
+                Guard.AgainstNegativeAndZero("value", value);
+                networkRecoveryInterval = value;
+            }
+        }
+
+        /// <summary>
+        /// Creates new instance of the RabbitMQ transport.
+        /// </summary>
+        public RabbitMQTransport(string connectionString) 
+            : base(TransportTransactionMode.ReceiveOnly, true, true, true)
+        {
+            if (connectionString.StartsWith("amqp", StringComparison.OrdinalIgnoreCase))
+            {
+                AmqpConnectionString.Parse(connectionString)(this);
```

Comment:
Having to call Parse and then cast like that? This is not something that should be required to do. Why not keep the `ConnectionConfiguration` class and not have all this going on?

Reply:
case? This is passing the current instance to the configuration action. Why having two classes with exactly same properties, one public and one private?